Cherry-picked SHOW command PRs (#3080)
neo4j/docs-operations: modules/ROOT/pages/configuration/show-settings.adoc

:description: Information about the `SHOW SETTINGS` command.
:keywords: SHOW SETTINGS, configuration settings, listing settings
:page-role: not-on-aura  new-5.6
= Show configuration settings

> This page describes listing configuration settings using the `SHOW SETTINGS` command.

Reviewer: This link will go to the current Cypher 25 version.

> For general information about the `SHOW` command, see the link:{neo4j-docs-base-uri}/cypher-manual/clauses/show/[Cypher Manual -> SHOW].

Reviewer: ```suggestion
For general information about the `SHOW` command, see the link:{neo4j-docs-base-uri}/cypher-manual/5/clauses/show/[Cypher Manual -> SHOW].
```

[NOTE]
====
* `SHOW SETTINGS` returns settings on the executing server only.
To retrieve settings on a specific server, you need to directly connect to it using `bolt` scheme.
For more information, see the link:{neo4j-docs-base-uri}/bolt/current/bolt[Bolt protocol documentation].

> * Configurations settings can also be listed using the xref:procedures/built-in-procedures.adoc#procedure_dbms_listConfig[`dbms.listConfig()`] procedure.

Reviewer 2: these bits weren't there before, but as long as they both exists in Cypher 5 as well it's no problem

should the link point to cypher 5 (Neo4j 5.26?) instead of current to stick to the version I'd assume we're on?
```suggestion
For more information, see the link:{neo4j-docs-base-uri}/bolt/5/bolt[Bolt protocol documentation].

* Configurations settings can also be listed using the xref:procedures/built-in-procedures.adoc#procedure_dbms_listConfig[`dbms.listConfig()`] procedure.
```

Author: There is no version 5 of that manual?

Reviewer 2: if there isn't then I guess it's fine 🤷  (I don't keep track of the manuals I'm not actively working in :P)

[options="header", width="100%", cols="1,5a"]
|===
| Action | Syntax

| List settings

a|
[source, syntax, role="noheader"]
----
SHOW SETTING[S] [setting-name[,...]]
[YIELD { * \| field[, ...] } [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
[WHERE expression]
[RETURN field[, ...] [ORDER BY field[, ...]] [SKIP n] [LIMIT n]]
----
|===

When using the `RETURN` clause, the `YIELD` clause is mandatory and must not be omitted.

[NOTE]
====
Setting names must be supplied as one or more comma-separated quoted `STRING` values or as an expression resolving to a `STRING` or a `LIST<STRING>`.
====

[[return-columns]]
== Return columns

The `SHOW SETTINGS` command returns the following columns:

.Listing settings output
[options="header", width="100%", cols="3m,6,2m,2a"]
|===
| Column
| Description
| Type
| Default output

| name
| The name of the setting.
| STRING
| {check-mark}

| value
| The current value of the setting.
| STRING
| {check-mark}

| isDynamic
| Whether the value of the setting can be updated dynamically, without restarting the server.

> For dynamically updating a setting value, see link:{neo4j-docs-base-uri}/operations-manual/current/configuration/dynamic-settings/[Update dynamic settings].

Reviewer: ```suggestion
For dynamically updating a setting value, see xref:configuration/dynamic-settings.adoc[Update dynamic settings].
```

| BOOLEAN
| {check-mark}

| defaultValue
| The default value of the setting.
| STRING
| {check-mark}

| description
| The setting description.
| STRING
| {check-mark}

| startupValue
| The value of the setting at last startup.
| STRING
|

| isExplicitlySet
| Whether the value of the setting is explicitly set by the user, either through configuration or dynamically.
| BOOLEAN
|

| validValues
| A description of valid values for the setting.
| STRING
|

| isDeprecated

> | Whether the setting is deprecated.

Reviewer 2: Missing the introduced in tag from the old Cypher 5 page
```suggestion
| Whether the setting is deprecated. label:new[Introduced in 5.9]
```

Reviewer: We put the labels at the beginning.

Author: Not in the Cypher docs 😄 
But I'll go with the Ops Manual convention here obviously, and we should probably agree on one format here (though outside the scope of this PR)

> | Whether the setting is deprecated.

Reviewer: ```suggestion
| label:new[Introduced in 5.9] Whether the setting is deprecated.
```

| BOOLEAN
|
|===

[[examples]]
== Examples

The following examples show how to use the `SHOW SETTINGS` command to list configuration settings.

=== List all settings with default return columns
[source, cypher]
----
SHOW SETTINGS
----

=== List all settings with all return columns

[source, cypher]
----
SHOW SETTINGS YIELD *
----

=== List all settings with specified return columns
[source, cypher]
----
SHOW SETTINGS YIELD name, defaultValue, startupValue
----

> === List named settings

Reviewer 2: we don't want to keep the text from this example?
```
It is possible to specify which settings to return in the list by setting names.
```
